### PR TITLE
Fix/re merge et 1331

### PR DESCRIPTION
**Pull request by moraleida**
### 🎫 Ticket

ET-1331 

### ✔️ Checklist
- [ ] I've included a changelog entry in the readme.txt file. 
- [ ] My code is tested. 
- [ ] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

### CODE REVIEW

File: src/Tickets/Commerce/Module.php [hunk 1]
```diff
@@ -439,7 +439,7 @@ public function supports_global_stock() {
 	 */
 
 	/**
-	 * Get's the product price html
+	 * Returns the ticket price html template
```
Comment (codingmusician) [inline src/Tickets/Commerce/Module.php:442]:
```suggestion
	 * Returns the ticket price html template.

File: src/Tickets/Commerce/Ticket.php [hunk 2]
```diff
@@ -128,6 +125,34 @@ class Ticket {
...
+	 * Stores the instance of the template engine that we will use for rendering the elements.
+	 *
+	 * @since TBD
+	 *
+	 * @var \Tribe__Template
+	 */
+	protected $template;
+
+	/**
+	 * Gets the template instance used to setup the rendering html
```
Comment (codingmusician) [inline src/Tickets/Commerce/Ticket.php:138]:
```suggestion
	 * Gets the template instance used to setup the rendering html.

File: src/Tickets/Commerce/Utils/Currency.php [hunk 3]
```diff
@@ -0,0 +1,416 @@
+<?php
+
+namespace TEC\Tickets\Commerce\Utils;
+
+/**
+ * Class Currency Utils
+ *
+ * This class holds stateless methods used to properly set-up currencies in Tickets Commerce
```
Comment (codingmusician) [inline src/Tickets/Commerce/Utils/Currency.php:8]:
```suggestion
 * This class holds stateless methods used to properly set-up currencies in Tickets Commerce.

File: src/Tickets/Commerce/Utils/Currency.php [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+	/**
+	 * Retrieve a fallback currency code.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public static function get_currency_code_fallback() {
+
+		// Check if we have a value set from Tribe Commerce
```
Comment (codingmusician) [inline src/Tickets/Commerce/Utils/Currency.php:64]:
```suggestion
		// Check if we have a value set from Tribe Commerce.

File: src/Tickets/Commerce/Utils/Currency.php [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+		$currency_code = tribe_get_option( static::$legacy_currency_code_option, static::$currency_code_fallback );
+
+		// Duplicate the currency code in the Tickets Commerce key.
+		tribe_update_option( static::$currency_code_option, $currency_code );
+
+		return $currency_code;
+	}
+
+	/**
+	 * Return the currency symbol to use as defined in the currency map
```
Comment (codingmusician) [inline src/Tickets/Commerce/Utils/Currency.php:74]:
```suggestion
	 * Return the currency symbol to use as defined in the currency map.

File: src/Tickets/Commerce/Utils/Currency.php [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+		 *
+		 * @param string $symbol The currency symbol.
+		 *
+		 * @return string
+		 */
+		return apply_filters( 'tec_tickets_commerce_currency_symbol', $symbol );
+	}
+
+	/**
+	 * Return the currency decimal separator character to use as defined in the currency map
```
Comment (codingmusician) [inline src/Tickets/Commerce/Utils/Currency.php:114]:
```suggestion
	 * Return the currency decimal separator character to use as defined in the currency map.

File: src/Tickets/Commerce/Utils/Currency.php [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+		 *
+		 * @param string $separator The currency decimal separator character.
+		 *
+		 * @return string
+		 */
+		return apply_filters( 'tec_tickets_commerce_currency_separator_decimal', $separator );
+	}
+
+	/**
+	 * Return the currency thousands separator character to use as defined in the currency map
```
Comment (codingmusician) [inline src/Tickets/Commerce/Utils/Currency.php:154]:
```suggestion
	 * Return the currency thousands separator character to use as defined in the currency map.

File: src/Tickets/Commerce/Utils/Currency.php [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+		 *
+		 * @param string $separator The currency thousands separator character.
+		 *
+		 * @return string
+		 */
+		return apply_filters( 'tec_tickets_commerce_currency_separator_thousands', $separator );
+	}
+
+	/**
+	 * Get and allow filtering of the currency symbol position
```
Comment (codingmusician) [inline src/Tickets/Commerce/Utils/Currency.php:194]:
```suggestion
	 * Get and allow filtering of the currency symbol position.

File: src/Tickets/Commerce/Utils/Currency.php [hunk 9]
```diff
@@ -0,0 +1,416 @@
...
+		 *
+		 * @since TBD
+		 *
+		 * @param string   $currency_position The currency position string.
+		 *
+		 * @return string
+		 */
+		$currency_position = apply_filters( 'tec_tickets_commerce_currency_symbol_position', $currency_position );
+
+		// Plugin's other code only accounts for one of these two values
```
Comment (codingmusician) [inline src/Tickets/Commerce/Utils/Currency.php:235]:
```suggestion
		// Plugin's other code only accounts for one of these two values.

File: src/Tickets/Commerce/Utils/Value.php [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+	public function set_up_currency_details() {
+		$this->currency_code                = Currency::get_currency_code();
+		$this->currency_symbol              = Currency::get_currency_symbol( $this->get_currency_code() );
+		$this->currency_symbol_position     = Currency::get_currency_symbol_position( $this->get_currency_code() );
+		$this->currency_separator_decimal   = Currency::get_currency_separator_decimal( $this->get_currency_code() );
+		$this->currency_separator_thousands = Currency::get_currency_separator_thousands( $this->get_currency_code() );
+	}
+
+	/**
+	 * Builds a list of Value objects from a list of numeric values
```
Comment (codingmusician) [inline src/Tickets/Commerce/Utils/Value.php:29]:
```suggestion
	 * Builds a list of Value objects from a list of numeric values.

File: src/Tickets/Commerce/Utils/Value.php [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+			if ( $value instanceof Value ) {
+				return $value;
+			}
+
+			return new self( $value );
+		}, $values );
+	}
+
+	/**
+	 * Get formatted html block with formatted currency and symbol
```
Comment (codingmusician) [inline src/Tickets/Commerce/Utils/Value.php:49]:
```suggestion
	 * Get formatted html block with formatted currency and symbol.

File: src/Tickets/Commerce/Utils/Value.php [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+
+		$position = 'prefix' === $this->get_currency_symbol_position() ? 'prefix' : 'postfix';
+
+		$html[] = "<span class'tribe-formatted-currency-wrap tribe-currency-{$position}'>";
+		$html[] = '<span class="tribe-currency-symbol">%1$s</span>';
+		$html[] = '<span class="tribe-amount">%2$s</span>';
+		$html[] = '</span>';
+
+		if ( $position !== 'prefix' ) {
+			// If position is not prefix, swap the symbol and amount span tags
```
Comment (codingmusician) [inline src/Tickets/Commerce/Utils/Value.php:65]:
```suggestion
			// If position is not prefix, swap the symbol and amount span tags.

File: src/views/v2/commerce/checkout/cart/footer/total.php [hunk 13]
```diff
@@ -7,34 +7,35 @@
...
- * @var array[]          $items                 [Global] List of Items on the cart to be checked out.
- * @var array[]          $gateways              [Global] An array with the gateways.
- * @var int              $gateways_active       [Global] The number of active gateways.
+ * @var \Tribe__Template                  $this            [Global] Template object.
+ * @var Module                            $provider        [Global] The tickets provider instance.
+ * @var string                            $provider_id     [Global] The tickets provider class name.
+ * @var array[]                           $items           [Global] List of Items on the cart to be checked out.
+ * @var array[]                           $gateways        [Global] An array with the gateways.
+ * @var int                               $gateways_active [Global] The number of active gateways.
+ * @var \TEC\Tickets\Commerce\Utils\Value $total_value     The total value of the cart
```
Comment (codingmusician) [inline src/views/v2/commerce/checkout/cart/footer/total.php:23]:
```suggestion
 * @var \TEC\Tickets\Commerce\Utils\Value $total_value     The total value of the cart.

File: src/views/v2/commerce/ticket/price.php [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * See more documentation about our views templating system.
+ *
+ * @link    https://evnt.is/1amp Help article for RSVP & Ticket template files.
+ *
+ * @since   TBD
+ *
+ * @version TBD
+ *
+ * @var \TEC\Tickets\Commerce\Utils\Value $price The Value instance of the ticket price
```
Comment (codingmusician) [inline src/views/v2/commerce/ticket/price.php:16]:
```suggestion
 * @var \TEC\Tickets\Commerce\Utils\Value $price The Value instance of the ticket price.

**Comment by moraleida** [comment]
All suggestion applied, this is ready for merge

**Review by bordoni — APPROVED** [review]

**Pull request merged**